core/src/test/java/com/google/cloud/sql/core/CoreSocketFactoryTest.java

```diff
@@ -364,6 +364,50 @@ public void testGetApplicationNameFailsAfterInitialization() {
         IllegalStateException.class, () -> CoreSocketFactory.setApplicationName("sample-app"));
   }
 
+  @Test
+  public void create_successfulNamedConnection() throws IOException, InterruptedException {
```

Comment:
These tests show how to use the `Connector` API, since CoreSocketFactory is the implementation of `Connector`

In real usage, an application would first call 
```
 ConnectionConfig real =
        new ConnectionConfig.Builder()
            .withCloudSqlInstance("myProject:myRegion:myInstance")
            .withIpTypes("PRIMARY")
            .build();
ConnectorRegistry.getConnector().register("my-connection", config);
```

And then to use the connection, they would use the JDBC URL: 
```
jdbc:mysql:///<DATABASE_NAME>?cloudSqlInstanceName=my-connection&socketFactory=com.google.cloud.sql.mysql.SocketFactory&user=<MYSQL_USER_NAME>&password=<MYSQL_USER_PASSWORD>
```